### ISSUE TITLE
Bug `no-unpublished-bin`: false positive with `files` starting with `./`

### ISSUE THREAD

**Issue by regseb** [eslint-community/eslint-plugin-n#99]
## Description

The [`no-unpublished-bin`](https://github.com/eslint-community/eslint-plugin-n/blob/bdfd61a/docs/rules/no-unpublished-bin.md) rule returns false positives when the paths of the `files` property (in `package.json`) begin with `"./"`.

## Files

- `package.json`

  ```JSON
  {
      "name": "testcase",
      "version": "1.0.0",
      "type": "module",
      "private": true,
      "bin": "./index.js",
      "files": ["./index.js"],
      "dependencies": {
          "eslint": "8.42.0",
          "eslint-plugin-n": "16.0.0"
      }
  }
  ```

- `eslint.config.js`

  ```JavaScript
  import n from "eslint-plugin-n";

  export default [
      {
          plugins: { n },
          rules: { "n/no-unpublished-bin": "error" },
      },
  ];
  ```

- `index.js`

  ```JavaScript
  console.log("Hello world");
  ```

## To reproduce

1. `npm install`
2. `npx eslint index.js`

```
/home/regseb/testcase/index.js
  1:1  error  npm ignores 'index.js'. Check 'files' field of 'package.json' or '.npmignore'  n/no-unpublished-bin

✖ 1 problem (1 error, 0 warnings)
```

## Additional information

With the value `{ "files": ["index.js"] }`, ESLint doesn't report any errors.

**Comment by regseb** [comment]
I don't reproduce the problem with _eslint-plugin-n_ `16.2.0`. This pull request https://github.com/eslint-community/eslint-plugin-n/pull/123 (which fixes https://github.com/eslint-community/eslint-plugin-n/issues/122) certainly corrected the problem.

**Comment by scagood** [comment]
Thank you for testing and confirming 😀
